src/PIL/ImageFont.py

```diff
@@ -562,9 +562,17 @@ def getmask2(
         _string_length_check(text)
         if start is None:
             start = (0, 0)
-        im, size, offset = self.font.render(
+        im = None
+
+        def fill(mode, size):
+            nonlocal im
+
+            im = Image.core.fill(mode, size)
```

Comment:
It might make sense to move the decompression bomb check here to simplify the C function.

Reply:
I've created https://github.com/python-pillow/Pillow/pull/7247 for this.